h2h/production-code/week6/get_data.py

```diff
@@ -4,85 +4,131 @@
         away win.
 
     Raises:
         None
     """
-    if read_filepath:
-        raw_dataset = pd.read_csv(read_filepath)
-        #when we read csv, a set will be interpreted as a string so we have to evaluate
+    if read_unencoded: #have unencoded csv, just need to encode
+        raw_dataset = pd.read_csv(read_unencoded)
         raw_dataset["winning_team_players"] = raw_dataset["winning_team_players"].apply(lambda x:literal_eval(x))
         raw_dataset["losing_team_players"] = raw_dataset["losing_team_players"].apply(lambda x:literal_eval(x))
-    else:
-        if write_filepath_unencoded:
-            raw_dataset = player_in_game_dataset(season_start, season_end, write_filepath = write_filepath_unencoded)
-        else:
-            raw_dataset = player_in_game_dataset(season_start, season_end)
+    else: #generate from scratch, decide if want to write the unencoded to CSV
+        raw_dataset = _generate_player_in_game_dataset(season_start, season_end, write_unencoded = write_unencoded)
 
     #want the names of all the players who played, because that will be our feature column names
     all_players = set()
-    for row in record.itertuples():
+    for row in raw_dataset.itertuples():
         all_players.update(row.winning_team_players,row.losing_team_players)
     all_players = sorted(list(all_players))
 
     #we condense all the names for ease, then add a label column, and create the dataframe
     cols = [condensed_name(player) for player in all_players]
     cols.append("label")
-    encoded_record = pd.DataFrame(columns=cols, index=range(record.shape[0]))
+    encoded_data = pd.DataFrame(columns=cols, index=range(raw_dataset.shape[0]))
     for row in raw_dataset.itertuples():
 
         #label is 1 for a home win, and home players have a 1 / away players have a -1
         if row.winning_team_id == row.home_team_id:
             for player in row.winning_team_players:
-                encoded_record.at[row.Index, condensed_name(player)] = 1
+                encoded_data.at[row.Index, condensed_name(player)] = 1
             for player in row.losing_team_players:
-                encoded_record.at[row.Index, condensed_name(player)] = -1
-            encoded_record.at[row.Index, "label"] = 1
+                encoded_data.at[row.Index, condensed_name(player)] = -1
+            encoded_data.at[row.Index, "label"] = 1
         #label is 0 for an away win, home players again have 1 / away players -1
         else:
             for player in row.winning_team_players:
-                encoded_record.at[row.Index, condensed_name(player)] = -1
+                encoded_data.at[row.Index, condensed_name(player)] = -1
             for player in row.losing_team_players:
-                encoded_record.at[row.Index, condensed_name(player)] = 1
-            encoded_record.at[row.Index, "label"] = 0
+                encoded_data.at[row.Index, condensed_name(player)] = 1
+            encoded_data.at[row.Index, "label"] = 0
```

Comment:
Instead of having duplicated code in an if/else, you can do something like
if ___:
    home_score = 1
else:
    home_score = -1
encoded_data.at[row.Index, condensed_name(player)] = home_score
...
encoded_data.at[row.Index, condensed_name(player)] = -home_score

Reply:
smart, thanks